src/elements/earn/portfolio/liquidityProtection/rewards/stake/RewardsStakeCTA.tsx

```diff
@@ -1,14 +1,68 @@
 import { Pool } from 'services/observables/tokens';
+import { stakeRewards } from 'services/web3/protection/rewards';
+import { useState } from 'react';
 
 interface Props {
   pool: Pool;
   account?: string | null;
+  errorBalance: string;
+  bntAmount: string;
 }
 
-export const RewardsStakeCTA = ({ pool, account }: Props) => {
+export const RewardsStakeCTA = ({
+  pool,
+  account,
+  errorBalance,
+  bntAmount,
+}: Props) => {
+  const [isBusy, setIsBusy] = useState(false);
+
+  const handleClick = async () => {
+    try {
+      setIsBusy(true);
+      const txHash = await stakeRewards({
+        amount: bntAmount,
+        poolId: pool.pool_dlt_id,
+      });
+    } catch (e) {
+      console.error('Staking Rewards failed with msg: ', e.message);
+    } finally {
+      setIsBusy(false);
+    }
+  };
+
+  const button = () => {
```

Comment:
I have a few problems with this:
1. the name of the function isn't descriptive and I need to read it to understand what it does and returns. We can have descriptive names and proper use of TS to make life easier
2. why invoke it again and again just to read different values from the result?? you actually call it 3 times to read 3 fields.